src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java

```diff
@@ -3145,4 +3155,131 @@ public ColumnModelPage getPossibleColumnModelsForViewScope(ViewScope scope, Stri
 		return getHtmlTeamMembers().contains(userId);
 	}
+	
+	@Override
+	public ProjectDisplayBundle getProjectDisplay(String projectId) throws RestServiceException {
+		try {
+			org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+			boolean isWiki = isWiki(projectId, synapseClient);
+			boolean isChallenge = isChallenge(projectId, synapseClient);
+			boolean isFile = isFileOrFolder(projectId, synapseClient);
+			boolean isTable = isTable(projectId, synapseClient);
+			boolean isDocker = isDocker(projectId, synapseClient);
+			boolean isForum = isForum(projectId, synapseClient);
+			return new ProjectDisplayBundle(isWiki, isFile, isTable, isChallenge, isForum, isDocker);
+		} catch (SynapseException e) {
+			throw ExceptionUtil.convertSynapseException(e);
+		}
+	}
+	
+	@Override
+	public boolean isChallenge(String projectId) throws RestServiceException {
+		try {
+			org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+			return isChallenge(projectId, synapseClient);
+		} catch (SynapseException e) {
+			throw ExceptionUtil.convertSynapseException(e);
+		}
+	}
+	
+	@Override
+	public boolean isForum(String projectId) throws RestServiceException {
+		try {
+			org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+			return isForum(projectId, synapseClient);
+		} catch (SynapseException e) {
+			throw ExceptionUtil.convertSynapseException(e);
+		}
+	}
+	
+	@Override
+	public boolean isDocker(String projectId) throws RestServiceException {
+		try {
+			org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+			return isDocker(projectId, synapseClient);
+		} catch (SynapseException e) {
+			throw ExceptionUtil.convertSynapseException(e);
+		}
+	}
+	
+	@Override
+	public boolean isFileOrFolder(String projectId) throws RestServiceException {
+		try {
+			org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+			return isFileOrFolder(projectId, synapseClient);
+		} catch (SynapseException e) {
+			throw ExceptionUtil.convertSynapseException(e);
+		}
+	}
+	
+	@Override
+	public boolean isTable(String projectId) throws RestServiceException {
+		try {
+			org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+			return isTable(projectId, synapseClient);
+		} catch (SynapseException e) {
+			throw ExceptionUtil.convertSynapseException(e);
+		}
+	}
+	
+	@Override
+	public boolean isWiki(String projectId) throws RestServiceException {
+		org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+		return isWiki(projectId, synapseClient);
+	}
+	
+	private boolean isForum(String projectId, org.sagebionetworks.client.SynapseClient synapseClient) throws SynapseException {
+		Forum forum = synapseClient.getForumByProjectId(projectId);
+		if (forum == null) {
+			return false;
+		}
+		return synapseClient.getThreadCountForForum(forum.getId(), DiscussionFilter.NO_FILTER).getCount() > 0;
```

Comment:
DiscussionFilter.NO_FILTER > what if there was a thread and a thread has been deleted? Assuming that the user delete the thread to be able to hide a tab? 

NO_FILTER allows retrieving both deleted and non-deleted thread. 
Suggest to use EXCLUDE_DELETED